[FEAT] CI & Swiftlint autocorrect 깃허브 액션 추가

0Hooni opened the pull request:
## 📌 이슈

- #93 

## ✅ 작업 사항

- [x] Swiftlint 적용
    - [x] lint rules 파일 추가
    - [x] 관련 세팅을 위한 빌드 파일 수정
- [x] CI 깃허브 액션 추가
- [x] Swiftlint autocorrect 깃허브 액션 추가
- [x] secrets.swift → Debug.xcconfig 수정 작업
    - [x] 빌드 세팅 수정
    - [x] .gitignore 수정
    - [x] 기존 코드가 변경된 키를 추적할 수 있도록 수정
- [x] 불필요한 파일, 타겟 제거
    - [x] 사용하지 않던 테스트 타겟(Unit test, UI test) 제거
    - [x] 디렉토리에 사용하지 않던 파일(AdminRepository/.swift) 제거


## 🚀 테스트 방식

### Swiftlint 테스트
- 노션에 프로젝트 세팅 문서의 "swiftlint 적용 과정"의 1번을 참고하여 swiftlint 설치
- 93번 브랜치에서 프로젝트 빌드

### 깃허브 액션 테스트
- 현재 PR을 오픈하여 적용된 깃허브 액션 적용 테스트 예정

### 변경된 Key 관리 테스트
- Swiftlint로 인해서 일반적인 상황으로는 빌드가 안됨
- 에러가 되는 린트 룰을 .swiftlint.yml 파일에서 비활성화 하여 빌드하여 테스트

## 👀 ETC (추후 개발해야 할 것, 참고자료 등) ->
> [!CAUTION]
> **이번 PR을 통해 Secrets.swift의 추적 제한이 풀렸습니다❗**
> **PR이 병합되고 난 후 반드시 Secrets.swift를 디렉토리에서 제거하고, 작업자의 커밋에 포함되지 않았는지 꼭 확인해주세요❗❗❗**

Madman-dev commented on PopPool/PopPool/Data/Network/PopPoolNetwork/UserDTO/InfoDTO/MyCommentInfoDTO.swift:
```diff
@@ -10,19 +10,19 @@ import Foundation
-            image: URL(string: image),
             likeCount: likeCount,
-            createDateTime: createDateTime
+            createDateTime: createDateTime.asDate(),
```
👍🏻

Madman-dev commented on PopPool/PopPool/Data/Network/PopPoolNetwork/CommentType.swift:
```diff
@@ -0,0 +1,13 @@
+//
+//  CommentType.swift
```
CommentType 형태로 인스타그램 구분 여부를 파악하기 위해 만든 것처럼 보이군요!
한가지 궁금한 점은 Instagram 등은 외부 데이터로 구분이 되어서일까요?
Data/Network에 넣은 이유는 무엇인가요??

Madman-dev commented on PopPool/PopPool/Domain/Interfaces/UserUseCase.swift:
```diff
@@ -20,21 +20,15 @@ protocol UserUseCase {
-        sort: [String]?
+        sort: [String]?,
+        commentType: CommentType
     ) -> Observable<GetMyCommentResponse>
```
👀

Madman-dev commented on PopPool/PopPool/Presentation/Scene/MyPage/MyCommentedPopUp/Cells/PopUpCommentedCell.swift:
```diff
@@ -115,15 +120,24 @@ extension PopUpCommentedCell: Cellable {
+        case .normal:
             setUpNormalComment()
-        } else {
+        case .instagram:
```
👍🏻👍🏻👍🏻

Madman-dev commented on PopPool/PopPool/Presentation/Scene/MyPage/MyCommentedPopUp/MyCommentedPopUpVM.swift:
```diff
@@ -12,25 +12,94 @@ import RxCocoa
+                    commentType: .instagram
+                )
+                .subscribe { response in
+                    owner.instaCommentList.accept(response)
```
behaviorRelay로 데이터를 계속해서 처리하는 형식일까요?
만약 맞다면 새로운 코멘트가 달릴 때마다 업데이트 되는 구조로 이해가 되는데요. 비용이 생각보다 많이 발생할 수 있을 가능성도 있을까요?

dongglehada replied:
behaviorRelay는 그 옵저버 객체를 관찰하는 것이지 서버의 변화를 관찰하고 있지는 않습니다.! API의 변화를 실시간으로 감지하는건 에초에 RESTAPI가 아닌 소켓통신을 이용해야 하는 것으로 알고 있습니다!

Madman-dev reviewed (commented):
코멘트 데이터를 받아 처리하는 BehaviorRelay 관련해서만 궁금합니다..!

zzangzzangguy commented:
확인했습니다

0Hooni committed:
remove/#93: 컨벤션 파일 삭제

0Hooni committed:
chore/#93: 빌드 파일 수정

- 컨벤션 파일 제거로 인한 빌드 파일 수정

0Hooni committed:
chore/#93: 프로젝트 세팅 수정

- Run SwiftLint 스크립트 추가
- SwiftLint rule 파일 추가

0Hooni committed:
add/#93: swiftlint rule 파일 추가

0Hooni committed:
chore/#93: SwiftLint rule 파일 기본 세팅

0Hooni committed:
remove#93: 불필요한 파일 삭제

0Hooni committed:
docs/#93: SwiftLint 문법 수정

0Hooni committed:
feat/#93: Swiftlint 자동 수정 깃허브 액션 추가

0Hooni committed:
remove/#93: 사용하지 않는 테스트 타겟 제거

0Hooni committed:
feat/#93: CI 깃허브 액션 추가

- Build 테스트 포함
- Swiftlint 테스트 포함

0Hooni committed:
fix/#93: CI가 깃액션 봇의 커밋은 무시하도록 수정

0Hooni committed:
chore/#93: Debug.xcconfig로 대체하기 위한 빌드 설정 수정

- Debug.xcconfig를 Resource 폴더에 추가함
- 해당 파일은 ignore로 걸러지도록 함
- info에 Key의 이름을 등록해두어 Bundle로 찾도록 설정

0Hooni committed:
chore/#93: gitignore 수정 및 정리

- 새로운 Debug.xcconfig를 무시하도록 수정
- 기존 ignore 설정을 보기 쉽게 정리

0Hooni committed:
refactor/#93: 변경 된 키 저장 방식에 맞게 키 호출 코드 수정

- Secrets.swift의 enum을 열도록 할 예정
- 해당 파일에서 Key 호출을 위한 여러 컴퓨티드 프로퍼티 제공
- 변경된 프로퍼티에 맞게 호출 코드도 일부 수정

0Hooni committed:
refactor/#93: GitHubActions 파일 추가 수정

- 깃헙 액션에 맞게 파일 이름 수정
- CI → ci
- Swiftlint autocorrect → swiftlint-autocorrect
- 추가로 autocorrect의 작동 시기를 Push → PR open으로 수정하여 깃액션 작동 빈도를 줄임

0Hooni committed:
chore/#93: CI에 Debug.xcconfig 생성 스텝 추가

0Hooni committed:
refactor/#93: Secrets.swift 제거

- Secrets enum을 대체할 KeyPath enum을 추가
- gitignore에서 Secrets.swift 추적 제한 제거
- 빌드 파일 수정

0Hooni committed:
refactor/#93: Secrets enum → KeyPath enum 변경 대응

0Hooni committed:
chore/#93: CI 태스크 이름 수정

- 몇몇 이모지가 생각보다 촌스러워서 수정함

0Hooni committed:
test/#93: CI 테스트를 위한 임시 swiftlint룰 적용

0Hooni commented:
<img width="800" alt="스크린샷 2025-04-03 오전 12 22 14" src="https://github.com/user-attachments/assets/f80bb2b1-c511-455d-8150-0c9331e3dc57" />

성공적인 첫 CI 🥹

0Hooni committed:
refactor/#93: CI, AutoCorrect 통합

0Hooni committed:
fix/#93: 린트 자동수정 명령어 수정

0Hooni committed:
fix/#93: 린트 자동수정 에러 수정 및 빌드파일 경로 탐지 추가

0Hooni committed:
fix/#93: 이슈번호 검출 정규표현식 개선 및 린트 자동수정 커밋 에러 해결

0Hooni committed:
fix/#93: 자동수정 커밋 액션 No upstream branch 에러 수정

github-actions[bot] committed:
style/#93: Apply SwiftLint autocorrect

0Hooni commented:
### 🥹 자동 수정이랑 빌드 테스트 전부 작동 테스트 완료 🥹

<img width="240" alt="스크린샷 2025-04-03 오전 1 11 19" src="https://github.com/user-attachments/assets/a23aed96-5bb8-4ee6-ba42-e861ff195b6b" />

### 자동 수정으로 줄어들은 warning 수(4000 → 100 실화?!?!?)

<img width="80" alt="스크린샷 2025-04-03 오전 1 12 53" src="https://github.com/user-attachments/assets/0071643a-fab7-4532-8bf1-40c7ac1dc6c2" />

0Hooni committed:
Merge branch 'dev' of https://github.com/PopPool/iOS into style/#93-apply-CI-with-swiftlint

# Conflicts:
#	Poppool/Poppool/Application/AppDelegate.swift
#	Poppool/Poppool/Presentation/Admin/AdminRegister/PopUpStoreRegisterViewController.swift
#	Poppool/Poppool/Presentation/Admin/Data/MapDomain/MapPopUpStore.swift
#	Poppool/Poppool/Presentation/Map/Common/GMSMapViewDelegateProxy.swift
#	Poppool/Poppool/Presentation/Map/Common/MapUtilities.swift
#	Poppool/Poppool/Presentation/Map/FillterSheetView/FilterBottomSheetView.swift
#	Poppool/Poppool/Presentation/Map/FillterSheetView/FilterBottomSheetViewController.swift
#	Poppool/Poppool/Presentation/Map/FindMap/MapGuideView/FullScreenMapViewController.swift
#	Poppool/Poppool/Presentation/Map/FindMap/MapGuideView/MapGuideViewController.swift
#	Poppool/Poppool/Presentation/Map/MapView/MapMarker.swift
#	Poppool/Poppool/Presentation/Map/MapView/MapView.swift
#	Poppool/Poppool/Presentation/Map/MapView/MapViewController.swift

Co-Authored-By: ki hyeon kim <creeps1104@gmail.com>

0Hooni committed:
refactor/#93: Swiftlint 에러 수정

github-actions[bot] committed:
style/#93: Apply SwiftLint autocorrect

0Hooni committed:
style/#93: 코드 스타일에 맞게 일부 코드 수정

Co-Authored-By: JunYoung <ghddns34@gmail.com>

dongglehada commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,102 @@
+            echo "No changes to commit"
+          fi
+
+  build:
```
Build Flow를 여기서 정의하는 걸까요..? 신기합니다!

0Hooni replied:
@dongglehada

ci 내부에 여러 jobs을 둘 수 있고 현재 총 두개의 job을 두고 있습니다. 
- autocorrect: 해당 job에 포함된 step에 따라 swiftlint 자동 수정 및 수정 내용 커밋
- build: 스텝에 따라 빌드 환경을 구성한 뒤 빌드와 린트 테스트

이 두가지의 작업이 성공적으로 끝나면 PR이 병합될 수 있는 상태로 판단해줍니다!

zzangzzangguy replied:
친절한 설명감사합니다.. 확인했습니다!

dongglehada commented on Poppool/.swiftlint.yml:
적용된 lint룰 확인하였습니다!

dongglehada commented on Poppool/Poppool/Resource/Info.plist:
Info 파일 수정도 확인하였습니다 :>

dongglehada reviewed (approved):
시크릿에서 키패스로 변경이 되었군요 변경상항은 확인 하였습니다! 정리해주신 내용 확인해보겠습니다!
